examples/inline-yaml.cf

```diff
@@ -0,0 +1,49 @@
+#+begin_src cfengine3
+bundle agent example_inline_yaml
+#@ brief Example illustrating inline yaml
+{
+  vars:
+      # YAML requires "---" header (followed by newline)
+      # NOTE \n is not interpreted as a newline, instead use $(const.n)
+
+      "yaml_multi_line" data => '---
+- key1: value1
+- key2: value2';
+
+
+      "yaml_single_line" data => '---$(const.n)- key1: value1$(const.n)- key2: value2';
+
+  reports:
+      "Data container defined from yaml_multi_line: $(with)"
+        with => string_mustache( "{{%-top-}}", @(yaml_multi_line) );
+
+      "Data container defined from yaml_single_line: $(with)"
+        with => string_mustache( "{{%-top-}}", @(yaml_single_line));
+}
+bundle agent __main__
```

Comment:
Do you have to use `__main__` for some reason? Otherwise I would just change the `example_inline_yaml` bundle name to `main`. For the sake of brevity I guess.

Reply:
I like to use `__main__` because it highlights the feature, and it makes it much easier to scoop up a bunch of examples into a policy and then run the bundles you want. If they are all `bundle agent main` then you have to go fix duplicate bundle definition.

Reply:
Makes sense. I typically only put an example stand-alone and run it with `cf-agent -KIf ./%`. :+1: on your method.

Reply:
I've always been a fan of self-contained examples I can run directly so I'm with Craig.

Although it would be neat if `cf-agent -f myfile.cf` ran whatever agent bundle it found if it was the only one.